src/compile.js

```diff
@@ -1809,7 +1810,7 @@ Compiler.prototype.cclass = function (s) {
     entryBlock = this.newBlock("class entry");
 
     this.u.prefixCode = "var " + scopename + "=(function $" + s.name.v + "$class_outer($globals,$locals,$rest){var $gbl=$globals,$loc=$locals;";
-    this.u.switchCode += "return(function " + s.name.v + "(){";
+    this.u.switchCode += "return(function " + fixReservedNames(s.name.v) + "(){";
```

Comment:
Yeah I had the same thought, therefore I haven't opened a PR yet. While tinkering around it seems that class names like Number or String do no cause any trouble. As @bzwheeler stated we should use a descriptive name.

Maybe we need to have a look at properties and functions defs too?